### PR TITLE
latest occupancy estimation

### PR DESCRIPTION
**Pull request by Casper-NS**

### CODE REVIEW

File: CentralHub.Api.Model/Responses/Measurements/GetLatestOccupancyResponse.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    [JsonConstructor]
+    public GetLatestOccupancyResponse(int? occupancy, bool success)
+    {
+        EstimatedOccupancy = occupancy;
+        Success = success;
+    }
+
+    public bool Success { get; }
+
+    public int? EstimatedOccupancy { get; }
```
Comment (Taoshix) [inline CentralHub.Api.Model/Responses/Measurements/GetLatestOccupancyResponse.cs:16]: Nullable int, interesting

File: CentralHub.Api/Controllers/MeasurementController.cs [hunk 2]
```diff
@@ -87,7 +87,29 @@ public async Task<GetAggregatedMeasurementsResponse> GetAggregateMeasurements(in
...
 
-    private static IReadOnlyList<AggregatedMeasurements> CreateMeasurements(IReadOnlyCollection<AggregatedMeasurementDto> aggregatedMeasurements)
+    [HttpGet("occupancy/latest")]
+    public async Task<GetLatestOccupancyResponse> GetLatestEstimatedOccupancy(
+        int roomId,
+        float wifiDevicesPerPerson,
+        float bluetoothDevicesPerPerson,
+        CancellationToken cancellationToken)
+    {
+        var aggregatedMeasurements = (await measurementRepository.GetAggregatedMeasurementsAsync(roomId, cancellationToken)).Last();
```
Comment (Taoshix) [inline CentralHub.Api/Controllers/MeasurementController.cs:97]: Redundant parenthesis?

File: CentralHub.Api/Controllers/MeasurementController.cs [hunk 3]
```diff
@@ -87,7 +87,29 @@ public async Task<GetAggregatedMeasurementsResponse> GetAggregateMeasurements(in
...
+
+        if (aggregatedMeasurements == null)
+        {
+            return GetLatestOccupancyResponse.CreateUnsuccessful();
+        }
+
+        var occupancy =
+            (int)Math.Round(
+            bluetoothDevicesPerPerson == 0 ? 0 : aggregatedMeasurements.BluetoothCount / bluetoothDevicesPerPerson +
+            wifiDevicesPerPerson == 0 ? 0 : aggregatedMeasurements.BluetoothCount / wifiDevicesPerPerson);
```
Comment (Taoshix) [inline CentralHub.Api/Controllers/MeasurementController.cs:107]: integer division? but i guess that is okay in this case, however, it will always floor

**Review by Taoshix — APPROVED** [review]
Only minor things i could point out here

File: CentralHub.Api.Model/Responses/Measurements/GetLatestOccupancyResponse.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+namespace CentralHub.Api.Model.Responses.AggregatedMeasurements;
+
+public sealed class GetLatestOccupancyResponse
+{
+    [JsonConstructor]
+    public GetLatestOccupancyResponse(int? occupancy, bool success)
+    {
+        EstimatedOccupancy = occupancy;
+        Success = success;
+    }
```
Comment (mads256h) [inline CentralHub.Api.Model/Responses/Measurements/GetLatestOccupancyResponse.cs:12]:
```suggestion
    public GetLatestOccupancyResponse(bool success, int? occupancy)
    {
        Success = success;
        EstimatedOccupancy = occupancy;
    }
```

File: CentralHub.Api/Controllers/MeasurementController.cs [hunk 5]
```diff
@@ -87,7 +93,50 @@ public async Task<GetAggregatedMeasurementsResponse> GetAggregateMeasurements(in
...
+            return (int)Math.Round(
+                aggregatedMeasurements.BluetoothCount / os.BluetoothDevicesPerPerson +
+                aggregatedMeasurements.WifiCount / os.WifiDevicesPerPerson);
+        }, cancellationToken);
+
+
+        return GetLatestOccupancyResponse.CreateSuccessful(occupancy);
+    }
+
+    [HttpPost("devices/set")]
```
Comment (mads256h) [inline CentralHub.Api/Controllers/MeasurementController.cs:117]:
```suggestion
    [HttpPost("settings/set")]
```

**Review by mads256h — CHANGES_REQUESTED** [review]
small stuff

File: CentralHub.Api/Controllers/MeasurementController.cs (hunk 5 above)
Comment (Casper-NS) [reply CentralHub.Api/Controllers/MeasurementController.cs:117]: fixed

File: CentralHub.Api.Model/Responses/Measurements/GetLatestOccupancyResponse.cs (hunk 4 above)
Comment (Casper-NS) [reply CentralHub.Api.Model/Responses/Measurements/GetLatestOccupancyResponse.cs:12]: fixed

File: CentralHub.Api/Controllers/MeasurementController.cs (hunk 3 above)
Comment (Casper-NS) [reply CentralHub.Api/Controllers/MeasurementController.cs]: pretty surte its floating point devision, so its fine

Comment (mads256h) [reply CentralHub.Api/Controllers/MeasurementController.cs]: Its not floating point division

**Pull request merged**